packages/x-data-grid-pro/src/tests/rows.DataGridPro.test.tsx

```diff
@@ -760,124 +811,122 @@ describe('<DataGridPro /> - Rows', () => {
         field: baselineProps.columns[0].field,
       });
     });
 
-    it('should update the focus when clicking from one cell to another', () => {
-      render(<TestCase rows={baselineProps.rows} />);
-      fireUserEvent.mousePress(getCell(1, 0));
+    it('should update the focus when clicking from one cell to another', async () => {
+      const { user } = render(<TestCase rows={baselineProps.rows} />);
+      await user.click(getCell(1, 0));
       expect(gridFocusCellSelector(apiRef)).to.deep.equal({
         id: baselineProps.rows[1].id,
         field: baselineProps.columns[0].field,
       });
-      fireUserEvent.mousePress(getCell(2, 1));
+      await user.click(getCell(2, 1));
       expect(gridFocusCellSelector(apiRef)).to.deep.equal({
         id: baselineProps.rows[2].id,
         field: baselineProps.columns[1].field,
       });
     });
 
-    it('should reset focus when clicking outside the focused cell', () => {
-      render(<TestCase rows={baselineProps.rows} />);
-      fireUserEvent.mousePress(getCell(1, 0));
+    it('should reset focus when clicking outside the focused cell', async () => {
+      const { user } = render(<TestCase rows={baselineProps.rows} />);
+      await user.click(getCell(1, 0));
       expect(gridFocusCellSelector(apiRef)).to.deep.equal({
         id: baselineProps.rows[1].id,
         field: baselineProps.columns[0].field,
       });
-      fireUserEvent.mousePress(document.body);
+      await user.click(document.body);
       expect(gridFocusCellSelector(apiRef)).to.deep.equal(null);
     });
 
-    it('should publish "cellFocusOut" when clicking outside the focused cell', () => {
+    it('should publish "cellFocusOut" when clicking outside the focused cell', async () => {
       const handleCellFocusOut = spy();
-      render(<TestCase rows={baselineProps.rows} />);
+      const { user } = render(<TestCase rows={baselineProps.rows} />);
       apiRef.current?.subscribeEvent('cellFocusOut', handleCellFocusOut);
-      fireUserEvent.mousePress(getCell(1, 0));
+      await user.click(getCell(1, 0));
       expect(handleCellFocusOut.callCount).to.equal(0);
-      fireUserEvent.mousePress(document.body);
+      await user.click(document.body);
       expect(handleCellFocusOut.callCount).to.equal(1);
       expect(handleCellFocusOut.args[0][0].id).to.equal(baselineProps.rows[1].id);
       expect(handleCellFocusOut.args[0][0].field).to.equal(baselineProps.columns[0].field);
     });
 
-    it('should not crash when the row is removed during the click', () => {
-      expect(() => {
-        render(
-          <TestCase
-            rows={baselineProps.rows}
-            onCellClick={() => {
-              apiRef.current?.updateRows([{ id: 1, _action: 'delete' }]);
-            }}
-          />,
-        );
-        const cell = getCell(0, 0);
-        fireUserEvent.mousePress(cell);
-      }).not.to.throw();
+    it('should not crash when the row is removed during the click', async () => {
```

Comment:
expect doesn't support async, so I just removed the expects, if something throws inside the test it will fail.